core/src/main/java/org/jruby/util/Numeric.java

```diff
@@ -511,6 +511,15 @@ public static boolean f_odd_p(ThreadContext context, IRubyObject i) {
      */
     public static long i_gcd(long x, long y) {
         long shift, uz, vz;
+        if (x == Long.MIN_VALUE) {
+            if (y == Long.MIN_VALUE)
+                throw new IllegalArgumentException("gcd() is too big");
```

Comment:
but if x == y here couldn't we simply `return x;` ? there wasn't any kind of explicit Java exception being thrown before

Reply:
by definition the greatest common divisor (gcd) of two or more integers, which are not all zero, is the largest **positive integer** that divides each of the integers.  (https://en.wikipedia.org/wiki/Greatest_common_divisor)

Long.MIN_VALUE => -922337203685477580**8**
Long.MAX_VALUE => 922337203685477580**7**

if x and y are Long.MIN_VALUE, we can't represent it as a positive number because
Math.abs(Long.MIN_VALUE) => -9223372036854775808


gcd(Long.MIN_VALUE, Long.MIN_VALUE)
IMO it should raise an exception because we can't represent a correct result

if we use x than the result will be a negative number which is unexpected

bignum math
-9223372036854775808.gcd(-9223372036854775808)
=>9223372036854775808
the result is positive for the same parameters
-9223372036854775808.i_gcd(-9223372036854775808)
=> -9223372036854775808

given the fact, that the previous algorithm worked this way (and have the same limitation), also Math.abs doesn't raise
Math.abs(Long.MIN_VALUE) => -9223372036854775808
I'll return to the original behaviour

note that this is an edge case and probably never happens in the real code

Reply:
yy, thanks - have realized that afterwards. not sure what really should happen, maybe an explicit exception is for the best. but its great that for now it will behave as before.